apps/ewallet/lib/ewallet/formatters/amount_formatter.ex

```diff
@@ -3,19 +3,10 @@ defmodule EWallet.AmountFormatter do
   A string formatter for amounts
   """
   def format(amount, subunit_to_unit) do
-    subunit_to_unit
-    |> to_decimals()
-    |> float_to_binary(amount / subunit_to_unit)
-    |> String.replace_trailing(".0", "")
-  end
-
-  defp float_to_binary(decimals, value) do
-    :erlang.float_to_binary(value, [:compact, {:decimals, decimals}])
-  end
+    Decimal.set_context(%Decimal.Context{Decimal.get_context() | precision: 38})
```

Comment:
Also, can we refactor to this and move it to `EWallet.Application`?

```ex
# apps/ewallet/lib/ewallet/application.ex
defmodule EWallet.Application do
  # ..
  def start(_type, _args) do
    import Supervisor.Spec

    set_decimal_context()

    # ..
  end

  defp set_decimal_context do
    Decimal.get_context()
    |> Map.put(:precision, 38)
    |> Decimal.set_context()
  end
end
```

The refactor to pipes is for readability. The move to `Application` is because `Decimal.set_context()` is setting the context globally so it would be nice to set at application level rather than at specific functions.

Reply:
Yeah, that's nice too 👍